src/components/CIPipelineN/TaskDetailComponent.tsx

```diff
@@ -0,0 +1,266 @@
+    }: {
+        formData: FormType
+        setFormData: React.Dispatch<React.SetStateAction<FormType>>
+        selectedTaskIndex: number
+        activeStageName: string
+        appId: number
+        formDataErrorObj: FormErrorObjectType
+        calculateLastStepDetail: (
+            isFromAddNewTask: boolean,
+            _formData: FormType,
+            activeStageName: string,
+            startIndex?: number,
+        ) => {
+            index: number
+            calculatedStageVariables: Map<string, VariableType>[]
+        }
+        setFormDataErrorObj: React.Dispatch<React.SetStateAction<FormErrorObjectType>>
+    } = useContext(ciPipelineContext)
+    const [pageState, setPageState] = useState(ViewType.FORM)
+    const validationRules = new ValidationRules()
+    const [configurationType, setConfigurationType] = useState<string>('GUI')
+    const [editorValue, setEditorValue] = useState<string>('')
+
+    const currentStepTypeVariable =
+        formData[activeStageName].steps[selectedTaskIndex].stepType === PluginType.INLINE
+            ? 'inlineStepDetail'
+            : 'pluginRefStepDetail'
+
+    useEffect(() => {
+        if (formData[activeStageName].steps[selectedTaskIndex].stepType === PluginType.PLUGIN_REF) {
+            setPageState(ViewType.LOADING)
+            getPluginDetail(formData[activeStageName].steps[selectedTaskIndex].pluginRefStepDetail.pluginId, appId)
+                .then((response) => {
+                    setPageState(ViewType.FORM)
+                    processPluginData(response.result)
+                })
+                .catch((error: ServerErrors) => {
+                    setPageState(ViewType.ERROR)
+                    showError(error)
+                })
+        }
+    }, [])
+
+    const processPluginData = (pluginData) => {
+        const _form = { ...formData }
+        if (_form[activeStageName].steps[selectedTaskIndex].pluginRefStepDetail.outputVariables?.length === 0) {
+            _form[activeStageName].steps[selectedTaskIndex].pluginRefStepDetail.outputVariables =
+                pluginData.outputVariables?.map((variable) => {
+                    variable.variableStepIndexInPlugin = variable.variableStepIndex
+                    delete variable.variableStepIndex
+                    return variable
+                })
+            if (_form[activeStageName].steps.length > selectedTaskIndex) {
+                calculateLastStepDetail(false, _form, activeStageName, selectedTaskIndex)
+            }
+        }
+        if (_form[activeStageName].steps[selectedTaskIndex].pluginRefStepDetail.inputVariables?.length === 0) {
+            _form[activeStageName].steps[selectedTaskIndex].pluginRefStepDetail.inputVariables =
+                pluginData.inputVariables?.map((variable) => {
+                    variable.variableStepIndexInPlugin = variable.variableStepIndex
+                    delete variable.variableStepIndex
+                    return variable
```

Comment:
The inline function can be extracted to a separate function and used at both outputVariables & inputVariables.